charms/katib-db-manager/src/charm.py

```diff
@@ -130,10 +144,110 @@ def _check_mysql(self):
             # Ensure we've got some data sent over the relation
             mysql_data["root_password"]
         except (IndexError, StopIteration, KeyError):
-            raise CheckFailed("Waiting for mysql connection information", WaitingStatus)
+            raise ErrorWithStatus("Waiting for mysql connection information", WaitingStatus)
 
         return mysql_data
 
+    def _check_and_report_k8s_conflict(self, error):
+        """Return True if error status code is 409 (conflict), False otherwise."""
+        if error.status.code == 409:
+            self.logger.warning(f"Encountered a conflict: {error}")
+            return True
+        return False
+
+    def _apply_k8s_resources(self, force_conflicts: bool = False) -> None:
+        """Apply K8S resources.
+
+        Args:
+            force_conflicts (bool): *(optional)* Will "force" apply requests causing conflicting
+                                    fields to change ownership to the field manager used in this
+                                    charm.
+                                    NOTE: This will only be used if initial regular apply() fails.
+        """
+        self.unit.status = MaintenanceStatus("Creating K8S resources")
+        try:
+            self.k8s_resource_handler.apply()
+        except ApiError as error:
+            if self._check_and_report_k8s_conflict(error) and force_conflicts:
+                # conflict detected when applying K8S resources
+                # re-apply K8S resources with forced conflict resolution
+                self.unit.status = MaintenanceStatus("Force applying K8S resources")
+                self.logger.warning("Apply K8S resources with forced changes against conflicts")
+                self.k8s_resource_handler.apply(force=force_conflicts)
+            else:
+                raise GenericCharmRuntimeError("K8S resources creation failed") from error
+        self.model.unit.status = MaintenanceStatus("K8S resources created")
+
+    def _get_check_status(self):
```

Comment:
Do we really need a separate function that we use only once?

Reply:
I needed to mock it in unit tests so I can parameterize the status

Reply:
Got it.